[Bug]: conky 1.20.2 fails to build with specific options combination

### What happened?

When trying to update conky to 1.20.2 with my build options, it fails to build.
Relevant cmake build options:
-DBUILD_IMLIB2=yes -DBUILD_LUA_CAIRO=no -DBUILD_LUA_IMLIB2=yes

Full cmake options:
cmake -C /var/tmp/portage/app-admin/conky-1.20.2/work/conky-1.20.2_build/gentoo_common_config.cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_ARGB=yes -DBUILD_X11=yes -DBUILD_XDAMAGE=yes -DBUILD_XDBE=yes -DBUILD_XFIXES=yes -DBUILD_XSHAPE=yes -DBUILD_XINPUT=yes -DBUILD_MOUSE_EVENTS=yes -DOWN_WINDOW=yes -DBUILD_APCUPSD=no -DBUILD_AUDACIOUS=no -DBUILD_BUILTIN_CONFIG=yes -DBUILD_CMUS=no -DBUILD_CURL=no -DBUILD_DOCS=no -DBUILD_EXTRAS=yes -DBUILD_HDDTEMP=no -DBUILD_HTTP=no -DBUILD_I18N=yes -DBUILD_IBM=no -DBUILD_ICAL=no -DBUILD_ICONV=yes -DBUILD_IMLIB2=yes -DBUILD_INTEL_BACKLIGHT=no -DBUILD_IOSTATS=yes -DBUILD_IPV6=yes -DBUILD_IRC=no -DBUILD_JOURNAL=no -DBUILD_LUA_CAIRO=no -DBUILD_LUA_IMLIB2=yes -DBUILD_LUA_RSVG=no -DBUILD_MATH=yes -DBUILD_MOC=no -DBUILD_MPD=no -DBUILD_MYSQL=no -DBUILD_NCURSES=yes -DBUILD_NVIDIA=no -DBUILD_OLD_CONFIG=yes -DBUILD_PORT_MONITORS=yes -DBUILD_PULSEAUDIO=yes -DBUILD_RSS=no -DBUILD_TESTS=no -DBUILD_WAYLAND=yes -DBUILD_WLAN=yes -DBUILD_XFT=yes -DBUILD_XINERAMA=yes -DBUILD_XMMS2=no -DDOC_PATH=/usr/share/doc/conky-1.20.2 -DMAINTAINER_MODE=no -DRELEASE=yes -DPython3_EXECUTABLE=/usr/bin/python3.11 -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_TOOLCHAIN_FILE=/var/tmp/portage/app-admin/conky-1.20.2/work/conky-1.20.2_build/gentoo_toolchain.cmake /var/tmp/portage/app-admin/conky-1.20.2/work/conky-1.20.2

Build error:
[31mFAILED: [0mlua/CMakeFiles/conky-cairo_imlib2_helper.dir/libcairo_imlib2_helper.cc.o 
/usr/bin/x86_64-pc-linux-gnu-g++ -DTOLUA_RELEASE -D_LARGEFILE64_SOURCE -D_POSIX_C_SOURCE=200809L -Dconky_cairo_imlib2_helper_EXPORTS -I/var/tmp/portage/app-admin/conky-1.20.2/work/conky-1.20.2/3rdparty/toluapp/include -I/usr/include/lua5.4 -I/var/tmp/portage/app-admin/conky-1.20.2/work/conky-1.20.2_build -I/var/tmp/portage/app-admin/conky-1.20.2/work/conky-1.20.2/lua -I/var/tmp/portage/app-admin/conky-1.20.2/work/conky-1.20.2/src  -march=alderlake -mtune=alderlake -mabm -mno-cldemote -mno-kl -mno-sgx -mno-widekl -mshstk --param=l1-cache-line-size=64 --param=l1-cache-size=32 --param=l2-cache-size=30720 -O2 -pipe -fwrapv -std=c++17 -fPIC -Wno-unused-parameter -Wno-cast-qual -Wno-error=pedantic -MD -MT lua/CMakeFiles/conky-cairo_imlib2_helper.dir/libcairo_imlib2_helper.cc.o -MF lua/CMakeFiles/conky-cairo_imlib2_helper.dir/libcairo_imlib2_helper.cc.o.d -o lua/CMakeFiles/conky-cairo_imlib2_helper.dir/libcairo_imlib2_helper.cc.o -c /var/tmp/portage/app-admin/conky-1.20.2/work/conky-1.20.2_build/lua/libcairo_imlib2_helper.cc
/var/tmp/portage/app-admin/conky-1.20.2/work/conky-1.20.2_build/lua/libcairo_imlib2_helper.cc:16:10: fatal error: cairo.h: No such file or directory
   16 | #include <cairo.h>
      |          ^~~~~~~~~
compilation terminated.

It looks like conky tries to build helper which depends both on imlib2 and cairo when imlib2 is enabled and cairo is disabled, and fails. I suspect that it was broken in commit https://github.com/brndnmtthws/conky/commit/45500b1439f8284e87aa715a0e1557133ddbd924. Conky 1.20.1 builds fine with same options.

### Version

1.20.2

### Which OS/distro are you seeing the problem on?

Gentoo

### Conky config

_No response_

### Stack trace

_No response_

### Relevant log output

_No response_

--- a/cmake/ConkyBuildOptions.cmake
+++ b/cmake/ConkyBuildOptions.cmake
@@ -210,10 +210,10 @@ dependent_option(BUILD_MOUSE_EVENTS "Enable mouse event support" true
   "Mouse event support requires Wayland or X11 enabled")
 
 # Lua library options
-dependent_option(BUILD_LUA_CAIRO "Build cairo bindings for Lua" false
+dependent_option(BUILD_LUA_CAIRO "Build Cairo bindings for Lua" false
   "BUILD_GUI" false
   "Cairo Lua bindings depend on BUILD_GUI")
-dependent_option(BUILD_LUA_CAIRO_XLIB "Build Imlib2 bindings for Lua" true
+dependent_option(BUILD_LUA_CAIRO_XLIB "Build Cairo & Xlib interoperability for Lua" true
   "BUILD_X11;BUILD_LUA_CAIRO" false
   "Cairo Xlib Lua bindings require Cairo and X11")
 dependent_option(BUILD_LUA_IMLIB2 "Build Imlib2 bindings for Lua" false

--- a/lua/CMakeLists.txt
+++ b/lua/CMakeLists.txt
@@ -79,7 +79,7 @@ if(BUILD_LUA_IMLIB2)
   print_target_properties(conky-imlib2)
 endif(BUILD_LUA_IMLIB2)
 
-if(BUILD_LUA_IMLIB2)
+if(BUILD_LUA_CAIRO AND BUILD_LUA_IMLIB2)
   include_directories(${luacairo_includes} ${luaimlib2_includes}
     ${CMAKE_CURRENT_SOURCE_DIR})
   wrap_tolua(luacairo_imlib2_helper_src cairo_imlib2_helper.pkg)
@@ -93,7 +93,7 @@ if(BUILD_LUA_IMLIB2)
     ${luaimlib2_libs}
     toluapp_lib_static)
   set(lua_libs ${lua_libs} conky-cairo_imlib2_helper)
-endif(BUILD_LUA_IMLIB2)
+endif(BUILD_LUA_CAIRO AND BUILD_LUA_IMLIB2)
 
 if(BUILD_LUA_RSVG)
   include_directories(${luarsvg_includes} ${CMAKE_CURRENT_SOURCE_DIR})

